### DISCUSSION TITLE
ios 15.5  Certificate verification error

### DISCUSSION THREAD

**Discussion by liuyingduo** [mitmproxy/mitmproxy#5437] [Intercept Everything!]
#### Problem Description
Ios15.5 has correctly installed the certificate and enabled the certificate trust setting, but the certificate verification error still occurs
#### Steps to reproduce the behavior:
1. brew install mitmproxy
2. Set machine up to proxy through mitmproxy
3. In simulator Safari, go to mitm.it and install certificate.
4. Install Profile in Setting
5. “General” -> “About” -> “Certificate Trust Settings”. Make sure the switch next to “mitmproxy” text is on
6. Get the error Certificate verification error for gdmf.apple.com: self signed certificate in certificate chain (errno: 19, depth: 2)  and Cannot establish TLS with client (sni: xp.apple.com): TlsException("SSL handshake error: SysCallError(-1, 'Unexpected EOF')")

#### System Information
Paste the output of "mitmproxy --version" here.
Mitmproxy: 8.1.0
Python:    3.9.13
OpenSSL:   OpenSSL 3.0.3  3 May 2022
Platform:  macOS-12.4-x86_64-i386-64bit

**Comment by anandcpanchal** [comment]
I confirm above steps to reproduce same issue on my end.
Earlier it was working with older iOS version.

**Comment by mhils** [comment]
Assuming that other domains work, you are running into [Certificate Pinning](https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning).
